lib/Resource.js

```diff
@@ -149,12 +162,12 @@ Resource.prototype._addRelation = function(relationName, rawRelation, resource)
 };
 Resource.prototype._removeRelation = function(relationName, rawRelation, resource) {
   this._checkIsResource(resource);
-  if (rawRelation.data instanceof Array) {
+  if (Resource._rawRelationHasMany(rawRelation)) {
     rawRelation.data = rawRelation.data.filter(function(id) {
-      return !resource._matchesId(id);
+      return !resource._matches(id);
     });
     this[relationName] = this[relationName].filter(function(id) {
-      return !resource._matchesId(id);
+      return !resource._matches(id);
```

Comment:
I'm surprised this has changed, how was it working before? Or wasn't it? Do we need a test for it?